controllers/kubeadmconfig_controller.go

```diff
@@ -160,7 +154,29 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		}
 	}()
 
-	if !cluster.Status.ControlPlaneInitialized {
+	switch {
```

Comment:
I was looking more for a switch to organize the quick-to-return cases before generating the cloud-init

What do you think about restructuring this a little bit to include just a few cases that return early like:

* is the cluster's infrastructure ready?
- if it's not ready then we should return
* is the config ready but the infrastructure not?
- refresh the token and return
* is the config ready and the infrastructure ready?
- we're done; return

Reply:
Oooh, yeah, I like that much more. I'll give it a shot.